tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -192,3 +197,95 @@ func ScaleMachinePoolNodes(client *rancher.Client, cluster *v1.SteveAPIObject, n
 
 	return scaledClusterResp, nil
 }
+
+// AddRKE2K3SCustomClusterNodes is a helper method that will add nodes to the custom RKE2/K3S custom cluster.
+func AddRKE2K3SCustomClusterNodes(client *rancher.Client, cluster *v1.SteveAPIObject, nodes []*nodes.Node, rolesPerNode []string) error {
+	clusterStatus := &apiv1.ClusterStatus{}
+	err := v1.ConvertToK8sType(cluster.Status, clusterStatus)
+	if err != nil {
+		return err
+	}
+
+	token, err := tokenregistration.GetRegistrationToken(client, clusterStatus.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	var command string
+	for key, node := range nodes {
+		logrus.Infof("Adding node %s to cluster %s", node.NodeID, cluster.Name)
+		if strings.Contains(rolesPerNode[key], "windows") {
+			command = fmt.Sprintf("powershell.exe %s -Address %s", token.InsecureWindowsNodeCommand, node.PublicIPAddress)
+		} else {
+			command = fmt.Sprintf("%s %s --address %s", token.InsecureNodeCommand, rolesPerNode[key], node.PublicIPAddress)
+		}
+
+		output, err := node.ExecuteCommand(command)
+		if err != nil {
+			return err
+		}
+
+		logrus.Infof(output)
+	}
```

Comment:
this all looks very similar to provisioning/creates.go. I would consider these functions part of provisioning CRUD operations, so I think this should live in `extensions/provisioning`. That would also make it easier to pull this function out into a helper instead of being re-defined again here.  Thoughts?

Reply:
Fair point, I agree after thinking over it. I'll move it over.